arbnode/execution/sequencer.go

```diff
@@ -369,8 +374,8 @@ func (s *Sequencer) onNonceFailureEvict(_ addressAndNonce, failure *nonceFailure
 
 var ErrRetrySequencer = errors.New("please retry transaction")
 
-func (s *Sequencer) ctxWithQueueTimeout(inctx context.Context) (context.Context, context.CancelFunc) {
-	timeout := s.config().QueueTimeout
+// ctxWithTimeout is like context.WithTimeout except a timeout of 0 means unlimited instead of instantly expired.
+func ctxWithTimeout(inctx context.Context, timeout time.Duration) (context.Context, context.CancelFunc) {
```

Comment:
Applied, thanks for the suggestion. By "same below" do you mean with `parentCtx` in `PublishTransaction`? There, I think it's important to name the different contexts, because we have quite a few of them and we wouldn't want to accidentally use the one named `ctx` without thinking about it.